protocol/messenger_backup.go

```diff
@@ -78,6 +78,118 @@ func (m *Messenger) startBackupLoop() {
 }
 
 func (m *Messenger) BackupData(ctx context.Context) (uint64, error) {
+	clock, chat := m.getLastClockWithRelatedChat()
+	contactsToBackup := m.backupContacts(ctx)
+	communitiesToBackup, err := m.backupCommunities(ctx, clock)
+	if err != nil {
+		return 0, err
+	}
+	profileToBackup, err := m.backupProfile(ctx, clock)
+	if err != nil {
+		return 0, err
+	}
+
+	// Update contacts messages encode and dispatch
+	for i, d := range contactsToBackup {
+		d.ContactsDetails = &protobuf.BackupDataDetails{
+			DataNumber:  uint32(i + 1),
+			TotalNumber: uint32(len(contactsToBackup)),
```

Comment:
So this looks a bit unintuitive. `contactsToBackup` is a single message that can actually contain multiple contacts.
In other words: `TotalNumber` might be `1` and `DateNumber` might be `1` as well, but the amount of contacts to backup is actually #n 

Not sure if it causes a problem here, just want to raise this to be sure..

Reply:
This mechanism packs messages pretty well and accurate.

Please check `backupContacts` in context of `BackupData` function, so we're preparing backup messages in a way that each of them contains up to `BackupContactsPerBatch` contacts. Now for example if there are 25 contacts and we're packing let's say 10 contacts per message, we will have 3 backup messages, 2 of them with 10 contacts and one with 5 contacts and backup messages will look in that case like this:
```
// Message 1
protobuf.Backup {
   Contacts: []*SyncInstallationContactV2 // <-----contains 10 contacts
  BackupDataDetails {
    DataNumber:  1,
    TotalNumber:  3
}

// Message 2
protobuf.Backup {
   Contacts: []*SyncInstallationContactV2 // <-----contains 10 contacts
  BackupDataDetails {
    DataNumber:  2,
    TotalNumber:  3
}

// Message 3
protobuf.Backup {
   Contacts: []*SyncInstallationContactV2 // <-----contains 5 contacts
  BackupDataDetails {
    DataNumber:  3,
    TotalNumber:  3
}
```